platforms/android/tangram/src/main/java/com/mapzen/tangram/networking/DefaultHttpHandler.java

```diff
@@ -29,61 +29,14 @@
 
     private OkHttpClient okClient;
 
-    /**
-     * Construct an {@code DefaultHttpHandler} with default options.
-     */
-    public DefaultHttpHandler() {
-        final OkHttpClient.Builder builder = new OkHttpClient.Builder()
-                .followRedirects(true)
-                .followSslRedirects(true)
-                .connectTimeout(10, TimeUnit.SECONDS)
-                .writeTimeout(10, TimeUnit.SECONDS)
-                .readTimeout(30, TimeUnit.SECONDS);
-
-
-        builder.socketFactory(new SocketFactory() {
-            SocketFactory factory = SocketFactory.getDefault();
-
-            @Override
-            public Socket createSocket() throws IOException {
-                Socket s = factory.createSocket();
-                try {
-                    //Log.d("Tangram", "Patching socket nodelay: " + s.getTcpNoDelay());
-                    s.setTcpNoDelay(true);
-                } catch (SocketException e) {
-                    // empty
-                }
-                return s;
-            }
-
-            @Override
-            public Socket createSocket(String s, int i) throws IOException, UnknownHostException {
-                return null;
-            }
-
-            @Override
-            public Socket createSocket(String s, int i, InetAddress inetAddress, int i1) throws IOException, UnknownHostException {
-                return null;
-            }
-
-            @Override
-            public Socket createSocket(InetAddress inetAddress, int i) throws IOException {
-                return null;
-            }
-
-            @Override
-            public Socket createSocket(InetAddress inetAddress, int i, InetAddress inetAddress1, int i1) throws IOException {
-                return null;
-            }
-        });
-
-        configureClient(builder);
-
-        okClient = builder.build();
-    }
-
     @Override
     public Object startRequest(@NonNull final String url, @NonNull final HttpHandler.Callback cb) {
+        // Build the OkHttpClient lazily. If we run configureClient() in the constructor then
+        // subclasses in Kotlin won't yet be initialized.
+        if (okClient == null) {
+            okClient = buildClient();
+        }
```

Comment:
```suggestion
        if (okClient == null) {
            synchronized {
                if (okClient == null) {
                    okClient = buildClient();
                }
            }
        }
```
(or synchronize on a lock object)

If an instance of this class should be usable by several threads at once (i.e. a threadpool), the lazy initialization would need to be threadsafe.

Reply:
Yeah good point - this class is definitely used by multiple threads within Tangram. I've updated the change.